services/user/Webmail/ui/src/components/nav.tsx

```diff
@@ -0,0 +1,96 @@
+    TooltipTrigger,
+} from "../shad/components/ui/tooltip";
+
+interface NavProps {
+    isCollapsed: boolean;
+    links: {
+        title: string;
+        label?: string;
+        icon: LucideIcon;
+        variant: "default" | "ghost";
+        href: string;
+    }[];
+}
+
+export function Nav({ links, isCollapsed }: NavProps) {
+    return (
+        <div
+            data-collapsed={isCollapsed}
+            className="group flex flex-col gap-4 py-2 data-[collapsed=true]:py-2"
+        >
+            <nav className="grid gap-1 px-2 group-[[data-collapsed=true]]:justify-center group-[[data-collapsed=true]]:px-2">
+                {links.map((link, index) =>
+                    isCollapsed ? (
+                        <Tooltip key={index} delayDuration={0}>
+                            <TooltipTrigger asChild>
+                                <a
+                                    href={link.href}
+                                    className={cn(
+                                        buttonVariants({
+                                            variant: link.variant,
+                                            size: "icon",
+                                        }),
+                                        "h-9 w-9",
+                                        link.variant === "default" &&
+                                            "dark:bg-muted dark:text-muted-foreground dark:hover:bg-muted dark:hover:text-white",
+                                    )}
+                                >
+                                    <link.icon className="h-4 w-4" />
+                                    <span className="sr-only">
+                                        {link.title}
+                                    </span>
+                                </a>
+                            </TooltipTrigger>
+                            <TooltipContent
+                                side="right"
+                                className="flex items-center gap-4"
+                            >
+                                {link.title}
+                                {link.label && (
+                                    <span className="ml-auto text-muted-foreground">
+                                        {link.label}
+                                    </span>
+                                )}
+                            </TooltipContent>
+                        </Tooltip>
+                    ) : (
+                        <a
+                            key={index}
+                            href={link.href}
+                            className={cn(
+                                buttonVariants({
+                                    variant: link.variant,
+                                    size: "sm",
+                                }),
+                                link.variant === "default" &&
+                                    "dark:bg-muted dark:text-white dark:hover:bg-muted dark:hover:text-white",
```

Comment:
Would variables work here? Just wondering if we can stop all use of hard coding colours like this and go all in on [variables](https://ui.shadcn.com/docs/theming#list-of-variables)

Reply:
Especially if we're gonna change our theming later on, would be awesome changing a bunch of config files across several apps rather than looking for class names we wrote.

Reply:
Some of these are variables, aren't they? Like `muted`? Or maybe I don't understand.
See: https://github.com/gofractally/psibase/pull/798/files#diff-d9598b721971bbf6eac63ac68007806de62deab4c4e6075b5e9e470dcbae3ccdR5-R36

Whatever the case, I'm fine with harmonizing color treatment across apps. This was just path of least resistance to something demoable.

Reply:
Can these be dynamic? Such that a `theme` app could be used to set the value of the theme variables across all system apps (or potentially even user apps that want to conform to style guidelines?)